server/controllers/registration.controller.js

```diff
@@ -0,0 +1,58 @@
+'use strict';
+
+const responseUtil = require('../utils/responseUtils');
+const MessageUtil = require('../utils/messageUtil');
+const userRegistrationService = require('../services/registration.service');
+const userServices = require('../services/user.service')
+const { sendCourseRegistrationMail } = require('../utils/emailUtils');
+const { v4: uuidv4 } = require('uuid');
+
+exports.registerUserToCourse = async (req, res) => {
+    const {
+        candidateName,
+        courseCode,
+        userId,
+    } = req.body;
+    const id = uuidv4();
+    try {
+        const courseData = {
+            id,
+            candidateName,
+            courseCode,
+            userId,
+            registrationStatus: "PENDING"
+        };
+        const userExists = await userServices.exists(userId);
+        if(userExists) {
+            const isUserErolledToCourse = await userRegistrationService.isUserAlreadyErolledToCourse(userId, courseCode);
```

Comment:
In `userRegistrationService.isUserAlreadyErolledToCourse(userId, courseCode)`,
I would prefer to take parameter `emailId` instead of `userId`, because we collect user data only when registration happens, so each time it will be a new `userId`.
also change `courseCode` to `courseId`
let me know what you think..

Reply:
Yes, I too think using email is a better option instead of userId.